testing/integration/agent_long_test.go

```diff
@@ -0,0 +1,282 @@
+		Description: "Test policy " + policyUUID,
+		MonitoringEnabled: []kibana.MonitoringEnabledOption{
+			kibana.MonitoringEnabledLogs,
+			kibana.MonitoringEnabledMetrics,
+		},
+	}
+
+	policyResp, err := tools.InstallAgentWithPolicy(ctx, runner.T(), installOpts, runner.agentFixture, runner.info.KibanaClient, basePolicy)
+	require.NoError(runner.T(), err)
+	// install system package
+
+	systemPackage := kibana.PackagePolicyRequest{}
+
+	jsonRaw, err := os.ReadFile("agent_long_test_base_system_integ.json")
+	require.NoError(runner.T(), err)
+
+	err = json.Unmarshal(jsonRaw, &systemPackage)
+	require.NoError(runner.T(), err)
+
+	systemPackage.ID = policyUUID
+	systemPackage.PolicyID = policyResp.ID
+	systemPackage.Namespace = "default"
+	systemPackage.Name = fmt.Sprintf("system-long-test-%s", policyUUID)
+	systemPackage.Vars = map[string]interface{}{}
+
+	runner.T().Logf("Installing fleet package....")
+	_, err = runner.info.KibanaClient.InstallFleetPackage(ctx, systemPackage)
+	require.NoError(runner.T(), err, "error creating fleet package")
+
+	// install apache
+
+	policyUUIDApache := uuid.New().String()
+	apachePackage := kibana.PackagePolicyRequest{}
+
+	jsonRaw, err = os.ReadFile("agent_long_test_apache_integ.json")
+	require.NoError(runner.T(), err)
+
+	err = json.Unmarshal(jsonRaw, &apachePackage)
+	require.NoError(runner.T(), err)
+
+	apachePackage.ID = policyUUIDApache
+	apachePackage.PolicyID = policyResp.ID
+	apachePackage.Namespace = "default"
+	apachePackage.Name = fmt.Sprintf("system-long-test-%s", policyUUIDApache)
+	apachePackage.Vars = map[string]interface{}{}
+
+	runner.T().Logf("Installing fleet package....")
+	_, err = runner.info.KibanaClient.InstallFleetPackage(ctx, apachePackage)
+	require.NoError(runner.T(), err, "error creating fleet package")
+
+}
+
+func (runner *ExtendedRunner) TestHandleLeak() {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Hour)
+	defer cancel()
+
+	testRuntime := os.Getenv("LONG_TEST_RUNTIME")
+	if testRuntime == "" {
+		testRuntime = "5m"
+	}
+
+	// because we need to separately fetch the PIDs, wait until everything is healthy before we look for running beats
+	require.Eventually(runner.T(), func() bool {
+		allHealthy := true
+		status, err := runner.agentFixture.ExecStatus(context.Background())
+		require.NoError(runner.T(), err)
+		for _, comp := range status.Components {
+			if comp.State != int(cproto.State_HEALTHY) {
+				allHealthy = false
+			}
+		}
+		return allHealthy
+	}, time.Minute*3, time.Second*20)
+
+	procs, err := sysinfo.Processes()
+	require.NoError(runner.T(), err)
+
+	handles := []processWatcher{}
+
+	// track running beats
+	// the `last 30s` metrics tend to report gauges, which we can't use for calculating a derivative.
+	// so separately fetch the PIDs
+	for _, proc := range procs {
+		info, err := proc.Info()
+		require.NoError(runner.T(), err)
+		if strings.Contains(info.Name, "beat") || strings.Contains(info.Name, "elastic-agent") {
```

Comment:
The output of elastic agent status contains the PIDs of the processes agent is supervising, you should be able to parse them out of the `elastic-agent status --output=json` output.

Alternatively you could return the PID from the `State` RPC https://github.com/elastic/elastic-agent/blob/b6c24c122af4ba668e2db32ee9b9f1fabf25fc6c/control_v2.proto#L142

That is a lot more reliable than relying on the process name.

Reply:
Huh, the test framework doesn't have it: https://github.com/elastic/elastic-agent/blob/08ac21ae93da9a41b0b1b0610e4a917e3278b062/pkg/testing/fixture.go#L1099

Maybe just was never added to the types the test unmarshal to, then

Reply:
This is the only reference to the PID I can find in the status output: 
```
            "message": "Healthy: communicating with pid '405681'",
```

Is that what you meant? I mean, we can extract that with regex, but that seems pretty awkward too, since I don't think we consider the message string to be a stable interface.

Reply:
Yes that's what meant, I forgot it only appeared in the message. You can look at how much work it'd be to pull out into a dedicated field, but the regex is likely fine.

The string technically isn't a stable interface, but it is unlikely to change, and this is more reliable than greping manually for the process names. This also lets you clearly identify the component name so we can tell which integration/module is causing the problem.